core/src/main/java/aima/core/search/basic/informed/RecursiveBestFirstSearch.java

```diff
@@ -52,27 +52,27 @@
 	@Override
 	public List<A> apply(Problem<A, S> problem) {
 		// return RBFS(problem, MAKE-NODE(problem.INITIAL-STATE), infinity)
-		return rbfs(problem, new SuccessorNode(nodeFactory.newRootNode(problem.initialState()), this::h),
+		return rbfs(problem, newRootNode(problem.initialState(), 0),
 				Double.POSITIVE_INFINITY).result();
 	}
 
 	// function RBFS(problem, node, f_limit) returns a solution, or failure and
 	// a new f-cost limit
 	public Result rbfs(Problem<A, S> problem, SuccessorNode node, double f_limit) {
 		// if problem.GOAL-TEST(node.STATE) then return SOLUTION(node)
-		if (searchController.isGoalState(node.n, problem)) {
-			return new Result(searchController.solution(node.n));
+		if (problem.isGoalState(node.state())) {

```

Comment:
True, I'll fix it. 
Also, there are still many algorithms including BreadthFirstSearch, that contain problem.isGoalState() and not searchController.isGoalState(), shouldn't they be fixed as well , or if I'm mistaken, there is a reason to keep them ?